grails-app/services/org/pih/warehouse/core/OrganizationService.groovy

```diff
@@ -44,8 +44,25 @@ class OrganizationService {
         return findOrCreateOrganization(name, code, [])
     }
 
+    Organization findOrganization(String code, String name) {
+        Organization organization = Organization.createCriteria().get {
+            eq("code", code)
+            ne("code", "")
+            isNotNull("code")
+        }
+        if (!organization) {
+            organization = Organization.createCriteria().list(max: 1) {
+                eq("name", name)
+                ne("name", "")
+                isNotNull("name")
+                order("dateCreated", "asc")
```

Comment:
The current implementation of findOrganization just replicates what already existing with the dynamic query. But we should ask Kelsey and Manon whether this is appopriate since it basically says "return a record if we match on code or name" but the name could match multiple records that aren't at all related to the code that was passed in. Do we want to constrain the results by the unique code as well and create new records if that doesn't match.

Reply:
This discussion was moved to ticket https://pihemr.atlassian.net/browse/OBGM-753